Feature/fetch plots

masafumimori opened the pull request:
In this PR
- modified Prettier settings
- add API mock to return mocked TRDL data
- draw map based on data returned from API
- show data returned from API in modal

I have some concerns:
1. Should I use `plot` instead of `tile` for the type and variable names? (now they are mixed so being consistent would be good)
2. I am not sure if the directory structure is right. For example, I added `mock-data.ts` under `pages/api/nfts/type` but should I create another folder containing only mock data?
3. Is it ok to have constants in helper files? I know that sometimes constants are written in the `constant` files/folders so I would like to check your preference about it.

lorenzopolidori commented on .prettierrc.json:
```diff
@@ -2,7 +2,8 @@
     "trailingComma": "none",
-    "bracketSameLine": true
+    "bracketSameLine": true,
+    "arrowParens": "avoid"
```
👍

lorenzopolidori commented on components/map/index.tsx:
```diff
@@ -31,21 +32,10 @@ const HORIZONTAL_SCROLL_SENSITIVITY = 0.05
-        10, 10, 10, 34, 20, 21, 29, 33, 33, 26, 33, 33, 33, 30, 20, 11, 22, 22,
-        22, 22, 22, 22, 22, 22, 12
-    ]
+    const [tileMaps, setTileMaps] = useState<TileMapType[]>([])
```
Maybe better `const [mapTiles, setMapTiles] = useState<MapTileType[]>([])`

lorenzopolidori commented on components/map/index.tsx:
```diff
@@ -65,14 +55,16 @@ const Map = ({ width, height, headerHeight, onSelectTile }: MapProps) => {
             for (let tileX = 0; tileX < GRID_SIZE; ++tileX) {
                 for (let tileY = 0; tileY < GRID_SIZE; ++tileY) {
-                    const imageIndex = tileMap[tileY * GRID_SIZE + tileX]
+                    const index = (tileY * GRID_SIZE + tileX) % 14
```
Why % 14?

masafumimori replied:
Sorry I forgot to delete this. I was using this when I was trying to see the API response works ( at the time only 15 elements existed)

lorenzopolidori commented on utils/api-config.js:
```diff
@@ -14,13 +14,15 @@ const terracellContract = (id, applicationId) => `/api/terracells/${id}/contract
 const nfts = accountId => `/api/accounts/${accountId}/nfts`
 const fileUpload = '/api/files/upload'
 const ipfsFiles = '/api/ipfs/files'
+const plots = (symbol, next) => `/api/nfts/type/${symbol}${next ? `?next=${next}` : ''}`
```
This should be:
```js
const plots = next => `/api/nfts/type/trld${next ? `?next=${next}` : ''}`
```
as plots already implies that we are fetching `TRLD` NFTs.

Even better, I suggest we get more explicit in what we are fetching, as in:
```js
const terralands = next => `/api/nfts/type/trld${next ? `?next=${next}` : ''}`
```

lorenzopolidori commented on pages/api/nfts/type/mock-data.ts:
```diff
@@ -0,0 +1,87 @@
+const assetList = [
```
👍

lorenzopolidori commented on pages/api/nfts/type/mock-data.ts:
```diff
@@ -0,0 +1,87 @@
+        symbol: 'TRLD',
+        url: 'ipfs://QmecxV7vrVEUTjvtu7VutqxhoYLFGtEENXvhBPAcJLSHxH',
+        offchainUrl: 'https://images.terragrids.org/map-tile-grass.png',
+        position: '0,0',
```
Let's change this to:
```
...
positionX: 0,
positionY: 0,
...
```
as this is how I ended up implementing the real API: https://github.com/terragrids/api/blob/dev/src/repository/token.repository.js#L77

lorenzopolidori commented:
I have noticed that when I click on a plot, the information does not refer to the plot I selected, see video attached.

https://user-images.githubusercontent.com/2437709/182816688-9e84f05f-79bf-44cf-a790-55ab0ab196d5.mov

masafumimori commented:
@lorenzopolidori 
Thank you for reviewing and answering my questions.

I fixed where was pointed out.

>I have noticed that when I click on a plot, the information does not refer to the plot I selected, see video attached.
For this, I think it was due to the code I forgot to delete, `% 14` part, as it limits to use only elements between index 0 ~ 14 so there should have been some mismatch between actual `plotMaps` data and rendered image. (within `onClick` function, index wasn't calculated with remainder (%14), so true `plotMaps` data was shown in modal)
I deleted `%14` and it seems working fine.
